internal/globaldefault/default.go

```diff
@@ -80,3 +79,23 @@ func SetupDefaultActivation(subshell subshell.SubShell, cfg DefaultConfigurer, r
 
 	return nil
 }
+
+// SetAskedOnce tracks whether it has been called before using the
+// configuration type. A response of true indicates that the config has been
+// updated. A response of false indicates that the config has not been updated.
+// If the configuration "set" call fails, this functions always returns true.
+func SetAskedOnce(cfg DefaultConfigurer) bool {
```

Comment:
It feels awkward to me that we're saving the state of this prompt in the `globaldefault` pkg whilst not actually doing anything around prompts in this pkg. I think this would be more at home inside the activate package. Leave the `globaldefault` package for setting up the global default, not for handling UX around it.